bridges/LeBonCoinBridge.php

```diff
@@ -14,176 +13,211 @@ class LeBonCoinBridge extends BridgeAbstract {
-						'Équipements Industriels' => 'equipement_industriels',
-						'Restauration - Hôtellerie' => 'restauration_hotellerie',
-						'Fournitures de Bureau' => 'fournitures_de_bureau',
-						'Commerces & Marchés' => 'commerces_marches',
-						'Matériel médical' => 'materiel_medical'
+					'MATERIEL PROFESSIONNEL' => array(
+						 'Tous' => '56',
+						 'Matériel Agricole' => '57',
+						 'Transport - Manutention' => '58',
+						 'BTP - Chantier Gros-oeuvre' => '59',
+						 'Outillage - Matériaux 2nd-oeuvre' => '60',
+						 'Équipements Industriels' => '32',
+						 'Restauration - Hôtellerie' => '61',
+						 'Fournitures de Bureau' => '62',
+						 'Commerces & Marchés' => '63',
+						 'Matériel Médical' => '64'
 					),
 					'SERVICES' => array(
-						'Tous' => '_services_',
-						'Prestations de services' => 'prestations_de_services',
-						'Billetterie' => 'billetterie',
-						'Évènements' => 'evenements',
-						'Cours particuliers' => 'cours_particuliers',
-						'Covoiturage' => 'covoiturage'
+						 'Tous' => '31',
+						 'Prestations de services' => '34',
+						 'Billetterie' => '35',
+						 'Evénements' => '49',
+						 'Cours particuliers' => '36',
+						 'Covoiturage' => '65'
 					),
 					'MAISON' => array(
-						'Tous' => '_maison_',
-						'Ameublement' => 'ameublement',
-						'Électroménager' => 'electromenager',
-						'Arts de la table' => 'arts_de_la_table',
-						'Décoration' => 'decoration',
-						'Linge de maison' => 'linge_de_maison',
-						'Bricolage' => 'bricolage',
-						'Jardinage' => 'jardinage',
-						'Vêtements' => 'vetements',
-						'Chaussures' => 'chaussures',
-						'Accessoires & Bagagerie' => 'accessoires_bagagerie',
-						'Montres & Bijoux' => 'montres_bijoux',
-						'Équipement bébé' => 'equipement_bebe',
-						'Vêtements bébé' => 'vetements_bebe'
+						 'Tous' => '18',
+						 'Ameublement' => '19',
+						 'Electroménager' => '20',
+						 'Arts de la table' => '45',
+						 'Décoration' => '39',
+						 'Linge de maison' => '46',
+						 'Bricolage' => '21',
+						 'Jardinage' => '52',
+						 'Vêtements' => '22',
+						 'Chaussures' => '53',
+						 'Accessoires & Bagagerie' => '47',
+						 'Montres & Bijoux' => '42',
+						 'Equipement bébé' => '23',
+						 'Vêtements bébé' => '54',
 					),
-					'AUTRES' => 'autres'
+					'AUTRES' => '37'
+				)
+			),
+			'o' => array(
+				'name' => 'Vendeur',
+				'type' => 'list',
+				'values' => array(
+					'Tous' => '',
+					'Particuliers' => 'private',
+					'Professionnels' => 'pro',
 				)
 			)
 		)
 	);
 
 	public function collectData(){
 
-		$category = $this->getInput('c');
-		if(empty($category)) {
-				$category = 'annonces';
+		$params = array(
+			'text' => $this->getInput('k'),
+			'region' => $this->getInput('r'),
+			'category' => $this->getInput('c'),
+			'owner_type' => $this->getInput('o'),
+		);
+
+		$url = self::URI . 'rechereche/?' . http_build_query($params);
+		$headers = get_headers($url);
```

Comment:
Instead of first cheking headers and then using ```file_get_contents``` that we no longer use, you can do ```$html = getContents($url) or returnServerError("...")```

Reply:
Ho, fine! I tried it but didn't manage to make it fails. That's why I use the headers.
I fix it